### ISSUE TITLE
[A11y]: Interactive story autoplay is an accessibility violation

### ISSUE THREAD

**Issue by Sidnioulz** [storybookjs/storybook#32247]
### Describe the bug

**See prior reports**
- https://github.com/storybookjs/storybook/pull/29018 
- https://github.com/storybookjs/storybook/discussions/17949
- https://github.com/storybookjs/storybook/discussions/25011
- https://github.com/storybookjs/storybook/discussions/23616

The fact that stories autoplay in Storybook is a violation of [WCAG SC 2.2.2 (A)](https://www.w3.org/WAI/WCAG22/Understanding/pause-stop-hide.html) and [WCAG SC 2.3.3 (AAA)](https://www.w3.org/WAI/WCAG22/Understanding/animation-from-interactions.html).

A proposed solution in a prior PR has been to store a global flag in `localStorage`. The PR could not directly be merged directly because there needs to be documentation, tests, and UI changes in stories with mount and in the a11y and interactions tab to accompany this change. The scope of changes is large enough to warrant a discussion through RFC first.

The RFC will be open soon. This issue exists because all prior reports were through discussions or PRs, and we need an issue to track this in the accessibility project.

### Reproduction link

yarn storybook:ui

### Reproduction steps

1. Enable prefers-reduced-motion in your browser
2. Open Storybook
3. Note how stories automatically play, causing non-essential non-user-triggered animation (SC 2.3.3)
4. While it's kind of possible to stop the autoplay from the interactions tab, there isn't a locally available mechanism to stop a playing story, right next to it, or prior to loading it (SC 2.2.2)

### System

```bash
n/a, 9.2.0 alpha
```

**Comment by Sidnioulz** [comment]
Blocked by https://github.com/storybookjs/storybook/discussions/32381

**Comment by Sidnioulz** [comment]
### Proposed implementation

cc. @ghengeveld @MichaelArestad

* We create a `storyAutoplay` global, with three potential values: 'always', 'never', and 'no-reduced-motion' (default value; all names TBC)
* We add a `Select` or menu `Button` in the interactions panel and/or in the toolbar to control the `storyAutoplay` global; maybe we can mix this up with the remount button in the toolbar?
* Storybook instance admins would be able to enforce a default value globally, or on specific stories
* The `no-reduced-motion` value would read reduced motion preferences to decide if autoplay should be enabled or not

**Comment by MichaelArestad** [comment]
I have a few thoughts here. I don't have a clear solution, but what I would like to see is almost the opposite of your suggestion.

First, some general related thoughts/opinions on behavior currently:
* Stories should autoplay for testing purposes
* Most of the time, stories probably should not autoplay for users as the interactions hijack the user controls resulting in some frustrating interactions. This is problematic as the user likely is expecting to see the end result of a set of interactions when navigating to the story.
* Stories with interactions have never worked well in docs partly because the stories in docs are not isolated in their iframe.

From a user standpoint, I would like to explore:
1. What does it look like if we treat stories with interactions (that hijack the UI) like audio or video media? Should the story be greyed out with a big play button?
2. Isolating stories in docs via iframe and not autoplaying them. We could use the same treatment as in #1 or explore alternatives.
3. Having an option/global/url param/whatever for test suites (SB test, etc) to automatically run those interactions.
4. Having an option/UI somewhere for a user to enable autoplay (if they want to quickly work through a series of stories with interactions)

**Comment by Sidnioulz** [comment]
Also @MichaelArestad, sometimes devs use autoplay to "initialise" a story, e.g. open a modal. So disabling all autoplay by default could push us towards splitting up `play` into `arrange` and `act` (to match the [AAA pattern](https://semaphore.io/blog/aaa-pattern-test-automation)). That's probably too radical of a change and I'd rather we didn't end up having to consider that.

We have a lot of use cases to account for:
| Autoplay value | Forced on story | Motion preference | Expected outcome |
| -------------- | --------------- | ----------------- | ---------------- |
| default        | ø               | ø				   | Autoplay         |
| default        | ø               | reduced		   | Don't autoplay   |
| always         | ø               | ø				   | Autoplay         |
| always         | ø               | reduced		   | Autoplay and cause a WCAG violation<br />&nbsp;*OR*<br /> hide entire story under a play button?                 |
| never          | ø               | ø				   | Don't autoplay   |
| never          | ø               | reduced		   | Don't autoplay   | 
| *              | always          | ø				   | Autoplay         |
| *              | always          | reduced		   | Autoplay and cause a WCAG violation<br />&nbsp;*OR*<br /> hide entire story under a play button?                 |
| *              | never           | ø				   | Don't autoplay   |
| *              | never           | reduced		   | Don't autoplay   |


When it comes to Storybook Test, stories play in the background anyway, so autoplay will not affect that.

Other affected surface area will be the interactions panel, and any addon panel that measures stuff about stories where the SB dev intended the story to autoplay (a11y addon, performance addon, etc.).

So I think we need to have a top-level control for playing stories. I kinda wanna connect the "Remount" button to the autoplay feature, but without replacing it altogether. I'm curious to see if a ButtonGroup with Remount+Play buttons together would make sense.

Possibly, if we don't autoplay because of reduced motion:
* a warning banner telling the user that the story is not yet in its intended state
* or a warning-coloured status pill on the play button to make it visible, like we have for the TagsFilter button when filters are applied (and we use tooltip copy to contextualise)
